chore(style-editor: Render style editor form based on developer configuration

KevinDavilaDotCMS opened the pull request:
https://github.com/user-attachments/assets/06c07042-8030-4c24-a8f6-28431c69b256


You can use this on our example to test:

```
    const bannerForm = defineStyleEditorSchema({
        contentType: 'Banner',
        sections: [
            {
                title: 'Typography',
                fields: [
                    styleEditorField.input({
                        id: 'my-text',
                        label: 'Texto',
                        inputType: 'text',
                        defaultValue: ''
                    }),
                    styleEditorField.dropdown({
                        id: 'the-size',
                        label: 'Size',
                        options: [
                            {
                                label: '18',
                                value: '18px'
                            },
                            {
                                label: '24',
                                value: '24px'
                            },
                            {
                                label: '32',
                                value: '32px'
                            },
                            {
                                label: '40',
                                value: '40px'
                            }
                        ],
                        placeholder: 'Select an option',
                    }),
                    styleEditorField.radio({
                        id: 'font-family',
                        label: 'Font Family',
                        options: [
                            {
                                label: 'Arial',
                                value: 'arial'
                            },
                            {
                                label: 'Helvetica',
                                value: 'helvetica'
                            }
                        ]
                    }),
                    styleEditorField.checkboxGroup({
                        id: 'type-settings',
                        label: 'Type settings',
                        options: [
                            {
                                label: 'Bold',
                                value: 'bold'
                            },
                            {
                                label: 'Italic',
                                value: 'italic'
                            },
                            {
                                label: 'Underline',
                                value: 'underline'
                            },
                            {
                                label: 'Strikethrough',
                                value: 'strikethrough'
                            }
                        ],
                        defaultValue: {
                            option1: false,
                            option2: false
                        },
                    })
                ]
            },
            {
                title: 'Banner',
                fields: [
                    styleEditorField.radio({
                        id: 'layout',
                        label: 'Leyaut',
                        columns: 2,
                        options: [
                            {
                                label: 'Left',
                                value: 'left',
                                imageURL: 'https://hips.hearstapps.com/hmg-prod/images/one-piece-orden-cronologico-peliculas-1677501143.jpg?crop=1xw:0.9375xh;center,top&resize=1200:*'
                            },
                            {
                                label: 'Right',
                                value: 'right',
                                imageURL: 'https://hips.hearstapps.com/hmg-prod/images/one-piece-orden-cronologico-peliculas-1677501143.jpg?crop=1xw:0.9375xh;center,top&resize=1200:*'
                            },
                            {
                                label: 'Center',
                                value: 'center',
                                imageURL: 'https://hips.hearstapps.com/hmg-prod/images/one-piece-orden-cronologico-peliculas-1677501143.jpg?crop=1xw:0.9375xh;center,top&resize=1200:*'
                            },
                            {
                                label: 'Overlap',
                                value: 'overlap',
                                imageURL: 'https://hips.hearstapps.com/hmg-prod/images/one-piece-orden-cronologico-peliculas-1677501143.jpg?crop=1xw:0.9375xh;center,top&resize=1200:*'
                            }
                        ],
                    }),
                ]
            }
        ]
    })

    useStyleEditorSchemas([bannerForm])
```


This PR fixes: #33810

kevindaviladev committed:
WIP - Need to split the form to components.

kevindaviladev committed:
Enhance UVE Style Editor: Introduce new input components and refactor form structure

- Added new components for dropdown, checkbox group, and radio button fields to improve modularity and reusability.
- Refactored the style editor form to utilize these new components, enhancing readability and maintainability.
- Updated SCSS styles for better alignment with design specifications.
- Introduced unit tests for new components to ensure functionality and reliability.

This update aims to streamline the form handling process and improve the overall user experience in the UVE style editor.

kevindaviladev committed:
Bring back sdk react package.json

kevindaviladev committed:
Removed comments

kevindaviladev committed:
Removed unused code and fixed tests

kevindaviladev committed:
Refactor UVE Style Editor: Replace hardcoded field types with constants

- Updated the style editor form template and component to utilize constants for field types instead of hardcoded strings, enhancing maintainability and reducing potential errors.
- Introduced a new  constant in the shared consts file to standardize field type references across the application.

zJaaal commented on core-web/libs/portlets/edit-ema/portlet/src/lib/edit-ema-editor/components/dot-uve-palette/components/dot-uve-style-editor-form/dot-uve-style-editor-form.component.ts:
```diff
@@ -0,0 +1,149 @@
+                const config = field.config;
+
+                switch (field.type) {
+                    case 'dropdown':
```
dont we have a mapping for this?

KevinDavilaDotCMS commented on core-web/libs/portlets/edit-ema/portlet/src/lib/edit-ema-editor/components/dot-uve-palette/components/dot-uve-style-editor-form/components/uve-style-editor-field-radio/uve-style-editor-field-radio.component.scss:
```diff
@@ -0,0 +1,65 @@
+
+.radio-image-label:hover {
+    border-color: $color-palette-surface-800;
+    box-shadow: 0 1px 3px 0 rgb(0 0 0 / 0.1);
```
Cool, now we are using one from there

KevinDavilaDotCMS commented on core-web/libs/portlets/edit-ema/portlet/src/lib/edit-ema-editor/components/dot-uve-palette/components/dot-uve-style-editor-form/dot-uve-style-editor-form.component.ts:
```diff
@@ -0,0 +1,149 @@
+    }
+
+    #getDropdownDefaultValue(config: StyleEditorFieldSchema['config']): string {
+        if (typeof config?.defaultValue === 'string') {
```
Good catch!
I add the trim to the defaultValue

The validation to not accept values like `'    '` must be in runtime, so we can add a message in runtime saying the defaultValue is an empty string. We can add that in the [validation ticket](https://github.com/dotCMS/core/issues/34119)

KevinDavilaDotCMS replied:
Also i think the [guard clause](https://refactoring.guru/replace-nested-conditional-with-guard-clauses) is better for this function than the ternary, for the readability

kevindaviladev committed:
Refactor UVE Style Editor: Streamline field option properties and enhance form handling

- Updated the style editor form component to use constants for field types, improving maintainability.
- Simplified the radio field option structure by removing width and height properties, focusing on essential attributes.
- Adjusted the HTML and SCSS for radio field components to enhance styling consistency and responsiveness.
- Updated unit tests to reflect changes in the radio field option structure, ensuring accuracy in functionality.

kevindaviladev committed:
Fixed broken unit test
